umake/frameworks/ide.py

```diff
@@ -138,12 +133,264 @@ def is_installed(self):
         # check path and requirements
         if not super().is_installed:
             return False
-        if not isfile(join(self.install_path, "eclipse")):
+        if not isfile(join(self.install_path, "bin", self.executable)):
             logger.debug("{} binary isn't installed".format(self.name))
             return False
         return True
 
 
+class Eclipse(BaseEclipse):
+    """The Eclipse Foundation Luna distribution."""
+    download_page_url = "https://www.eclipse.org/downloads/download.php?" \
+                        "file=/technology/epp/downloads/release/luna/R/" \
+                        "eclipse-standard-luna-R-linux-gtk{arch}.tar.gz{suf}" \
+                        "&r=1"
+    executable = 'eclipse.sh'
+
+    def __init__(self, category):
+        super().__init__(name="Eclipse",
+                         description=_("Pure Eclipse Luna (4.4)"),
+                         category=category, only_on_archs=['i386', 'amd64'],
+                         download_page=self.download_page_url,
+                         dir_to_decompress_in_tarball='eclipse',
+                         desktop_filename='eclipse.desktop',
+                         packages_requirements=['openjdk-7-jdk'])
+
+
+class EclipseJava(BaseEclipse):
+    """The Eclipse Foundation Mars for Java Developers distribution."""
+    download_page_url = "https://www.eclipse.org/downloads/download.php?" \
+                        "file=/technology/epp/downloads/release/mars/R/" \
+                        "eclipse-java-mars-R-linux-gtk{arch}.tar.gz{suf}" \
+                        "&r=1"
```

Comment:
As we are going to only support eclise mars I guess, please factorize the whole url in BaseEclipse class, and then, we only change the URL for flavor. Making sense?